### PR TITLE
C converter cleanup

### PR DESCRIPTION
**Pull request by nerzhul**
Simple PR, remove unused functions, merge identical function & merge a single called function into the caller (it will never be used like this we never use double for pos, but float. Just kept this to keep the same functional way)

### CODE REVIEW

**Comment by nerzhul** [comment]
if i remember there is one regression in the code. I should split the code cleanup from the code merge

**Comment by Quent42340** [comment]
Ah? Yep could be nice then

File: src/script/common/c_converter.cpp [hunk 1]
```diff
@@ -300,16 +253,32 @@ void push_v3s16(lua_State *L, v3s16 p)
...
+	CHECK_POS_COORD("y");
+	pf.Y = lua_tonumber(L, -1);
+	CHECK_FLOAT_RANGE(pf.Y, "y")
+	lua_pop(L, 1);
+
+	lua_getfield(L, index, "z");
+	CHECK_POS_COORD("z");
+	pf.Z = lua_tonumber(L, -1);
+	CHECK_FLOAT_RANGE(pf.Z, "z")
+	lua_pop(L, 1);
```
Comment (SmallJoker) [inline src/script/common/c_converter.cpp]: Just re-use `check_v3f` here

File: src/script/common/c_converter.cpp [hunk 2]
```diff
@@ -300,16 +253,32 @@ void push_v3s16(lua_State *L, v3s16 p)
 
 v3s16 read_v3s16(lua_State *L, int index)
```
Comment (SmallJoker) [inline src/script/common/c_converter.cpp]: The naming is no longer consistent `check_v3s16` would fit much better here.

**Comment by SmallJoker** [comment]
Any progress?

**Comment by nerzhul** [comment]
Soo many problems on this MR closing.

**Pull request closed without merging**